spec/controllers/group/register_controller_spec.rb

```diff
@@ -0,0 +1,133 @@
+# encoding: utf-8
+
+#  Copyright (c) 2012-2013, Jungwacht Blauring Schweiz. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+require 'spec_helper'
+
+describe Group::RegisterController do
+
+  let(:group) { groups(:top_group) }
+  let(:person) { people(:top_leader) }
+
+
+  context 'with feature disabled' do
+    before do
+      expect(Settings.groups.self_registration).to receive(:activated).and_return(false)
+      group.update(self_registration_role_type: Group::TopGroup::Member.sti_name)
+    end
+
+    describe 'GET new' do
+      it 'redirects to group' do
+        sign_in(person)
+
+        get :new, params: { group_id: group.id }
+
+        is_expected.to redirect_to(group_path(group.id))
+      end
+    end
+
+    describe 'POST create' do
+      it 'redirects to group' do
+        sign_in(person)
+
+        post :create, params: {
+          group_id: group.id,
+          new_person: { email: 'foo@example.com' }
+        }
+
+        is_expected.to redirect_to(group_path(group.id))
+      end
+    end
+  end
+
+  context 'with feature enabled' do
+    before do
+      allow(Settings.groups.self_registration).to receive(:activated).and_return(true)
+    end
+
+    context 'GET new' do
+      context 'when registration active' do
+
+        before do
+          group.update(self_registration_role_type: Group::TopGroup::Member.sti_name)
+        end
+
+        context 'when unautorized' do
+          it 'renders page' do
+            get :new, params: { group_id: group.id }
+
+            is_expected.to render_template('group/register/new')
+          end
+        end
+
+        context 'when autorized' do
+          it 'renders page' do
+            sign_in(person)
```

Comment:
shouldn't this result in a redirect or access denied if a already logged in person tries to access this?